### PR TITLE
Dk login functionality

### PR DESCRIPTION
**Pull request by DavidKnott**
Fixes the agreement checkbox so that both our tests and a user must check it off before they login
Add in Google Oauth
Add in Twilio Verification
(Both are tested)
The Twilio test is skipped because it relies on information from the home index page that's needs to be integrated in
In the phone verification controller the SMS sender is skipped because twilio only gives under 100 free texts.  (I will work on stubbing it out)
A lot of styling still needs to be done.
I pulled down master and dealt with a merge conflict in our routes (just needed to delete header)

@jmejia 
@cews7 @Laszlo-JFLMTCO @ethanbennett

### CODE REVIEW

File: lib/code_generator.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module CodeGenerator
```
Comment (Laszlo-JFLMTCO) [inline lib/code_generator.rb:1]: @DavidKnott Great job breaking it out into its own method within the module, so we can reuse anywhere!

File: lib/confirmation_sender.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module ConfirmationSender
```
Comment (Laszlo-JFLMTCO) [inline lib/confirmation_sender.rb:1]: @DavidKnott I would suggest to refactor all two factor authentication related methods into a two factor authentication service. Similar to what we have done in class for calling external APIs.
What you do you think?
Adding @cews7 @ethanbennett

Comment (DavidKnott) [reply lib/confirmation_sender.rb:1]: Laszlo I'm open to refactoring this but I'd prefer to keep it as a module as opposed to a service class.  What do you guys think? @cews7 @ethanbennett

**Comment by Laszlo-JFLMTCO** [comment]
@DavidKnott @cews7 @ethanbennett Based on our team review we close this PR as it is covered under another PR.

**Pull request closed without merging**